examples/with-stripe-typescript/app/components/CheckoutForm.tsx

```diff
@@ -23,26 +35,57 @@ export default function CheckoutForm(): JSX.Element {
       [e.currentTarget.name]: e.currentTarget.value,
     });
 
+  const handleSubmit: React.FormEventHandler<HTMLFormElement> = async (
```

Comment:
Is there a reason this isn't a Server Action?

Reply:
`createCheckoutSession` _is_ a Server Action. I just can't pass it as a form action because there's logic dependent on props (`uiMode`).

Reply:
But if you're passing `uiMode` on a hidden input, couldn't you read that value off `FormData` instead?

Reply:
It's not clear to me how that would prevent the need for the event handler. Can you share some example code that demonstrates whet you're describing?

Dependent on the value of `uiMode`, we perform different actions:
- For `embedded` sessions we need to store `client_secret` in state returned from the Server Action and mount the session via Stripe.js.
- For `hosted` sessions we redirect to the `url` field returned from the the Server Action. Normally this can be done server-side, but in this case we're just using the JSON returned from the Server Action.

Reply:
Refactored the component to use the `action` prop.